hazelcast/src/main/java/com/hazelcast/config/security/LdapAuthenticationConfig.java

```diff
@@ -228,85 +232,85 @@ public LdapAuthenticationConfig setUserSearchScope(LdapSearchScope userSearchSco
         setIfConfigured(props, "roleSearchScope", roleSearchScope);
         setIfConfigured(props, "userNameAttribute", userNameAttribute);
 
-        if (useSystemUser) {
+        if (!isNullOrEmpty(systemUserDn)) {
             props.setProperty(Context.SECURITY_AUTHENTICATION, "simple");
             props.setProperty(Context.SECURITY_PRINCIPAL, systemUserDn);
             setIfConfigured(props, Context.SECURITY_CREDENTIALS, systemUserPassword);
             setIfConfigured(props, "passwordAttribute", passwordAttribute);
             setIfConfigured(props, "userContext", userContext);
             setIfConfigured(props, "userFilter", userFilter);
             setIfConfigured(props, "userSearchScope", userSearchScope);
+            setIfConfigured(props, "skipAuthentication", skipAuthentication);
         }
+        return props;
+    }
+
+    @Override
+    public LoginModuleConfig[] asLoginModuleConfigs() {
+        boolean useSystemUser = !isNullOrEmpty(systemUserDn);
+        LoginModuleConfig loginModuleConfig = new LoginModuleConfig(
+                useSystemUser ? "com.hazelcast.security.loginimpl.LdapLoginModule"
+                        : "com.hazelcast.security.loginimpl.BasicLdapLoginModule",
+                LoginModuleUsage.REQUIRED);
+
+        loginModuleConfig.setProperties(initLoginModuleProperties());
 
         return new LoginModuleConfig[] { loginModuleConfig };
     }
 
     @Override
     public String toString() {
-        return "LdapAuthenticationConfig [url=" + url + ", socketFactoryClassName=" + socketFactoryClassName + ", systemUserDN="
-                + systemUserDn + ", systemUserPassword=***, parseDN=" + parseDn + ", roleContext="
+        return "LdapAuthenticationConfig [url=" + url + ", socketFactoryClassName=" + socketFactoryClassName + ", systemUserDn="
+                + systemUserDn + ", systemUserPassword=" + systemUserPassword + ", parseDn=" + parseDn + ", roleContext="
```

Comment:
Why do we print a secret `systemUserPassword` here?

Reply:
Nice catch. I'll fix it. It was caused by regenerating the `toString` method in the Eclipse IDE.